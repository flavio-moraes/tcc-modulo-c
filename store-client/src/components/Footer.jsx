import { Facebook, Instagram, Twitter, YouTube } from "@material-ui/icons";
import { Link } from "react-router-dom";
import styled from "styled-components";

const Container = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  background-color: #dcecf1;
  --myw: 100vw;
`;
Container.displayName = "Footer";

const Left = styled.div`
  width: 300px;
  display: flex;
  flex-direction: column;
  padding: 20px;
`;

const Logo = styled.h1`
  margin: 8px 0px 20px;
`;

const Desc = styled.div`
  margin: 0px;
  font-size: small;
`;

const SocialContainer = styled.div`
  display: flex;
`;

const SocialIcon = styled.div`
  width: 40px;
  height: 40px;
  color: gray;
  display: flex;
  align-items: center;
  justify-content: center;
  margin-right: 16px;
`;

const Center = styled.div`
  padding: 20px;
  --value: calc(100vw - 900px);
  --binary: calc(
    clamp(0px, calc(var(--value) + 1px / 1000000), calc(1px / 1000000)) *
      1000000
  );
  width: calc(300px + var(--binary) * 100);
`;

const Title = styled.h3`
  margin: 20px 0px;
  width: 300px;
`;

const ListContainer = styled.div`
  display: flex;
  flex-wrap: wrap;
`;

const ListItem = styled.div`
  width: 180px;
  margin-bottom: 10px;
  margin-right: 20px;
`;

const Right = styled.div`
  width: 300px;
  padding: 20px;
`;

const PayMethod = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  width: 200px;
`;

const PayMethodItem = styled.img`
  margin-right: 8px;
  margin-top: 6px;
`;

const Footer = () => {
  return (
    <Container>
      <Left>
        <Logo>Loja Virtual</Logo>
        <Desc>
          <p style={{ margin: "0px" }}>
            Copyright © Loja Virtual S.A. 2021-2023
          </p>
          <p style={{ margin: "8px 0px" }}>Av Dom José Gaspar, 500</p>
          <p style={{ margin: "8px 0px" }}>
            CEP 30535-901 - Belo Horizonte/MG <br />
            CNPJ 12.345.678/9999-00
          </p>
        </Desc>
        <SocialContainer>
          <SocialIcon>
            <Facebook />
          </SocialIcon>
          <SocialIcon>
            <Instagram />
          </SocialIcon>
          <SocialIcon>
            <Twitter />
          </SocialIcon>
          <SocialIcon>
            <YouTube />
          </SocialIcon>
        </SocialContainer>
      </Left>
      <Center>
        <Title>Links Úteis</Title>
        <ListContainer>
          <ListItem>
            <Link to="/">Página inicial</Link>
          </ListItem>
          <ListItem>
            <Link to="/carrinho">Carrinho de compras</Link>
          </ListItem>
          <ListItem>
            <Link to="/conta/dados">Minha conta</Link>
          </ListItem>
          <ListItem>
            <Link to="/conta/pedidos">Meus pedidos</Link>
          </ListItem>
          <ListItem>
            <Link to="/conta/favoritos">Lista de desejos</Link>
          </ListItem>
          <ListItem>Contato</ListItem>
          <ListItem>
            <Link to="/sobre">Quem somos</Link>
          </ListItem>
        </ListContainer>
      </Center>
      <Right>
        <Title>Formas de Pagamento</Title>
        <PayMethod>
          <PayMethodItem src="/img/visa.svg" />
          <PayMethodItem src="/img/mastercard.svg" />
          <PayMethodItem src="/img/amex.svg" />
          <PayMethodItem src="/img/dinersclub.svg" />
          <PayMethodItem src="/img/hipercard.svg" />
          <PayMethodItem src="/img/elo.svg" />
          <PayMethodItem src="/img/boleto.svg" />
        </PayMethod>
      </Right>
    </Container>
  );
};

export default Footer;
