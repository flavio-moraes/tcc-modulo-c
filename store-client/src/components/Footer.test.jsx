import React from "react";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import Footer from "./Footer";

describe("Footer component", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<Footer />);

    expect(shallowToJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find("p").at(0).text()).toContain("Loja Virtual S.A.");
  });
});
