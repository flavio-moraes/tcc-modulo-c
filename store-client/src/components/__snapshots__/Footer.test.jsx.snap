// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders correctly 1`] = `
<Footer>
  <styled.div>
    <styled.h1>
      Loja Virtual
    </styled.h1>
    <styled.div>
      <p
        style={
          Object {
            "margin": "0px",
          }
        }
      >
        Copyright © Loja Virtual S.A. 2021-2023
      </p>
      <p
        style={
          Object {
            "margin": "8px 0px",
          }
        }
      >
        Av Dom José Gaspar, 500
      </p>
      <p
        style={
          Object {
            "margin": "8px 0px",
          }
        }
      >
        CEP 30535-901 - Belo Horizonte/MG 
        <br />
        CNPJ 12.345.678/9999-00
      </p>
    </styled.div>
    <styled.div>
      <styled.div>
        <Memo(ForwardRef(FacebookIcon)) />
      </styled.div>
      <styled.div>
        <Memo(ForwardRef(InstagramIcon)) />
      </styled.div>
      <styled.div>
        <Memo(ForwardRef(TwitterIcon)) />
      </styled.div>
      <styled.div>
        <Memo(ForwardRef(YouTubeIcon)) />
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.h3>
      Links Úteis
    </styled.h3>
    <styled.div>
      <styled.div>
        <Link
          to="/"
        >
          Página inicial
        </Link>
      </styled.div>
      <styled.div>
        <Link
          to="/carrinho"
        >
          Carrinho de compras
        </Link>
      </styled.div>
      <styled.div>
        <Link
          to="/conta/dados"
        >
          Minha conta
        </Link>
      </styled.div>
      <styled.div>
        <Link
          to="/conta/pedidos"
        >
          Meus pedidos
        </Link>
      </styled.div>
      <styled.div>
        <Link
          to="/conta/favoritos"
        >
          Lista de desejos
        </Link>
      </styled.div>
      <styled.div>
        Contato
      </styled.div>
      <styled.div>
        <Link
          to="/sobre"
        >
          Quem somos
        </Link>
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.h3>
      Formas de Pagamento
    </styled.h3>
    <styled.div>
      <styled.img
        src="/img/visa.svg"
      />
      <styled.img
        src="/img/mastercard.svg"
      />
      <styled.img
        src="/img/amex.svg"
      />
      <styled.img
        src="/img/dinersclub.svg"
      />
      <styled.img
        src="/img/hipercard.svg"
      />
      <styled.img
        src="/img/elo.svg"
      />
      <styled.img
        src="/img/boleto.svg"
      />
    </styled.div>
  </styled.div>
</Footer>
`;
